### PR TITLE
Implement single file upload management

### PR DESCRIPTION
**Pull request by cp-dharti-r**
> PR Description
- Implement a single file upload management component with a preview feature
- Allow uploading of the file by selecting or dragging and dropping the file
- Update `README.md` to mentioned workflow of the project

> Test Steps
- Check the `README.md` on https://github.com/canopas/react-file-upload/tree/single-file-upload#file-management-with-preview
- Checkout this branch and install the dependencies
- Run the dev server by `npm run dev`
- If you face any CSS issues then run `npm run build-css` after running the server

### CODE REVIEW

File: src/app/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { Inter } from "next/font/google";
+import "./globals.css";
+import "../../dist/css/output.css";
+import "../../dist/css/style.scss";
+
+const inter = Inter({ subsets: ["latin"] });
+
+export const metadata: Metadata = {
+  title: "React file upload management",
+  description: "Generated by create next app",
```
Comment (cp-sumi-k) [inline src/app/layout.tsx]: Let's add proper description of the library

File: src/app/page.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      isDragging: boolean;
+      isLoading: boolean;
+    }
+  );
+
+  const getPreviewFileData = (file: any) => {
+    setPreviewFileData(file);
+  };
+
+  const [uploadingStatus, setUploadingData] = useState(false);
```
Comment (cp-sumi-k) [inline src/app/page.tsx]: Both should be `[uploadingStatus, setUploadingStatus]`

File: src/app/page.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  const [getFile, setFileData] = useState({});
+
+  const getFileData = (file: any) => {
+    setFileData(file);
+  };
+
+  return (
+    <main className="flex min-h-screen flex-col justify-between p-5">
+      <SingleFileUpload
+        getPreview={getPreviewFileData}
```
Comment (cp-sumi-k) [inline src/app/page.tsx]: Make sure property names are same on both vue and react if possible

File: src/app/page.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        getFileObj={getFileData}
+        isUploading={isUploading}
+        uploadingStatus={uploadingStatus}
+        uploadBtn={"Save"}
+        progressBtn={"Saving..."}
+      >
+        <div className="m-5">
+          <div className="flex items-center justify-center">
+            {!fileData || !fileData.previewUrl ? (
+              <label className="flex flex-col items-center justify-center w-full h-56 border-2 border-gray-300 border-dashed rounded-lg cursor-pointer bg-gray-50 dark:hover:bg-bray-800 dark:bg-gray-700 hover:bg-gray-100 dark:border-gray-600 dark:hover:border-gray-500 dark:hover:bg-gray-600">
```
Comment (cp-sumi-k) [inline src/app/page.tsx:40]: Let's use [this](https://tailwindcss.com/blog/automatic-class-sorting-with-prettier), and sort classes for better readability

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -1 +1,268 @@
...
+
+```
+yarn add @canopassoftware/vue-file-upload
+```
+
+---
+
+## Examples
+
+We are providing some examples with design. so, you can easily take it and use into your project.
```
Comment (cp-sumi-k) [inline README.md]: We are providing some examples with design. so, you can easily use it in your project.

File: next.config.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  images: {
+    domains: ["picsum.photos"],
```
Comment (cp-sumi-k) [inline next.config.js]: Do we have used picsum photos anywhere?
If yes, let's download and add it to our server, so project will not have external dependency.

File: src/components/singleFile.tsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    };
+    reader.readAsDataURL(file);
+  };
+
+  const uploadingFunction = async () => {
+    setIsUploading(true);
+
+    await callback(fileObj);
+
+    console.log(uploadedFile)
```
Comment (cp-sumi-k) [inline src/components/singleFile.tsx]: Let's remove log if not required

File: src/components/singleFile.tsx [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+      previewFile(file);
+    }
+  };
+
+  const previewFile = (file: any) => {
+    fileData.previewType = "image";
+    const reader = new FileReader();
+
+    reader.onload = () => {
+      if (file.type.startsWith("image/")) {
```
Comment (cp-sumi-k) [inline src/components/singleFile.tsx:62]: Can make this function common, if its reusable

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

File: src/components/singleFile.tsx (hunk 8 above)
Comment (cp-dharti-r) [reply src/components/singleFile.tsx:62]: will refactor while implement for multiple file

**Review by cp-sumi-k — APPROVED** [review]

**Pull request merged**